core/gui/src/app/workspace/service/report-print/report-print.service.ts

```diff
@@ -0,0 +1,109 @@
+import { Injectable } from "@angular/core";
+import html2canvas from "html2canvas";
+import * as joint from "jointjs";
+import { Observable, Observer } from "rxjs";
+import { WorkflowActionService } from "../workflow-graph/model/workflow-action.service";
+import { WorkflowResultService } from "../workflow-result/workflow-result.service";
+
+@Injectable({
+  providedIn: "root",
+})
+export class ReportPrintService {
+  private paper!: joint.dia.Paper;
+
+  // Define constants for styles
+  private readonly SNAPSHOT_CONTAINER_WIDTH = "75%";
+  private readonly SNAPSHOT_CONTAINER_HEIGHT = "75%";
+  private readonly SNAPSHOT_CONTAINER_POSITION = "fixed";
+  private readonly SNAPSHOT_CONTAINER_TOP = "0";
+  private readonly SNAPSHOT_CONTAINER_LEFT = "0";
+  private readonly SNAPSHOT_CONTAINER_ZINDEX = "-1";
+  private readonly SNAPSHOT_CONTAINER_BACKGROUND = "#FFFFFF";
+  private readonly SCALE_FACTOR = 0.75;
+
+  constructor(
+    public workflowActionService: WorkflowActionService,
+    private workflowResultService: WorkflowResultService
+  ) {}
+
+  /**
+   * Captures the current workflow snapshot and returns it as a base64-encoded PNG image URL.
+   * @returns {Observable<string>} An Observable that emits the base64-encoded PNG image URL of the workflow snapshot.
+   */
+  public getWorkflowSnapshot(): Observable<string> {
+    return new Observable((observer: Observer<string>) => {
+      const element = document.querySelector("#workflow-editor") as HTMLElement;
+      if (element) {
+        const snapshotContainer = document.createElement("div");
+        snapshotContainer.style.width = this.SNAPSHOT_CONTAINER_WIDTH;
+        snapshotContainer.style.height = this.SNAPSHOT_CONTAINER_HEIGHT;
+        snapshotContainer.style.position = this.SNAPSHOT_CONTAINER_POSITION;
+        snapshotContainer.style.top = this.SNAPSHOT_CONTAINER_TOP;
+        snapshotContainer.style.left = this.SNAPSHOT_CONTAINER_LEFT;
+        snapshotContainer.style.zIndex = this.SNAPSHOT_CONTAINER_ZINDEX;
+        snapshotContainer.style.background = this.SNAPSHOT_CONTAINER_BACKGROUND;
+        document.body.appendChild(snapshotContainer);
+
+        const jointGraph = this.workflowActionService.getJointGraphWrapper().jointGraph;
+        const paper = new joint.dia.Paper({
+          el: snapshotContainer,
+          model: jointGraph,
+          width: snapshotContainer.clientWidth,
+          height: snapshotContainer.clientHeight,
+          interactive: false,
+          background: { color: "#FFFFFF" },
+        });
+
+        paper.scale(this.SCALE_FACTOR, this.SCALE_FACTOR);
+
+        // setTimeout is used to delay the execution of the code to ensure that the previous scaling operation has been completed.
```

Comment:
Why the scaling operator is asynchronized? are there any documentations about this operation? Generally, we should avoid using setTimeout. If an operation is async, then try to block until its finish to make it sync